internal/testdata/metric.go

```diff
@@ -197,12 +197,12 @@ func initHistogramMetric(hm pmetric.Metric) {
 	hdp1.SetSum(15)
 	hdp1.SetMin(15)
 	hdp1.SetMax(15)
-	hdp1.SetBucketCounts(pcommon.NewImmutableUInt64Slice([]uint64{0, 1}))
+	hdp1.SetBucketCounts(pcommon.NewUInt64SliceFromRaw([]uint64{0, 1}))
```

Comment:
Can we do something like:

```
hdp1.BucketCounts().FromRaw([]uint64{0, 1}))
```

Reply:
Makes sense. Let me try that

Reply:
Hmm.. We don't have such pattern in pdata in other places. If we replace `pcommon.Value.SetBytesVal` with `pcommon.Value.BytesVal().FromRaw(...)`, we also need to change the pattern for `Map` and `Slice` types of `pcommon.Value` where we currently use another approach: `pcommon.NewMapFromRaw(...).CopyTo(dest)`.

We can probably replace all the `New...FromRaw` with New...().CopyFromRaw(...)`:

```go
hdp1.BucketCounts().CopyFromRaw([]uint64{0, 1}))
val.MapVal().CopyFromRaw(...)
```



WDYT?

Reply:
I don't care,  `CopyFromRaw` sounds good, and better explain what will happen. On the minus side, FromRaw hides the "copy" so in the future is we find a solution to not copy we can optimize that. But result is the same.

Reply:
But I 100% agree about Map API to be similar with this.

Reply:
It improves the workflow for the use case with the buckets but makes it worse for Map->bytes use cases. 

E.g. instead of 
m.InsertBytes("key_bytes", pcommon.NewValueBytesFromRaw([]byte{1, 2, 3}))

user will need to have

Reply:
To reach consistency with `pcommon.Map` and `pcommon.Slice`, we will need to deprecate/remove `Value.<Insert/Upsert>Bytes` methods. 
But it's only 6 such calls in contrib. So maybe that's ok

Reply:
Oh and `pcommon.Value.SetBytesVal` should be removed as well. Other 6 usages in core

Reply:
Another downside of making the slices a "complex" type, is that we will need to store a pointer to the slice in https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/generated_immutable_slice.go#L23. So another extra allocation